packages/amplication-client/src/Application/code-view/CodeViewExplorer.tsx

```diff
@@ -0,0 +1,96 @@
+import { gql, useQuery } from "@apollo/client";
+import React, { useEffect, useState } from "react";
+import { App, Build, SortOrder } from "../../models";
+import BuildSelector from "../../Components/BuildSelector";
+import { FileDetails } from "./CodeViewPage";
+import { NodeTypeEnum } from "./NodeTypeEnum";
+import CodeViewExplorerTree from "./CodeViewExplorerTree";
+import "./CodeViewBar.scss";
+
+const CLASS_NAME = "code-view-bar";
+
+type Props = {
+  app: App;
+  onFileSelected: (selectedFile: FileDetails | null) => void;
+};
+
+export class FileMeta {
+  type!: NodeTypeEnum;
+  name!: string;
+  path!: string;
+  children?: FileMeta[] | undefined;
+  expanded?: boolean;
+}
+
+const CREATED_AT_FIELD = "createdAt";
+type TData = {
+  builds: Build[];
+};
+
+const CodeViewExplorer = ({ app, onFileSelected }: Props) => {
+  const [selectedBuild, setSelectedBuild] = useState<Build | null>(null);
+
+  const { data } = useQuery<TData>(GET_BUILDS_COMMIT, {
+    variables: {
+      appId: app.id,
+      orderBy: {
+        [CREATED_AT_FIELD]: SortOrder.Desc,
+      },
+    },
+    onCompleted: async (data) => {
+      setSelectedBuild(data.builds[0]);
+    },
+  });
+
+  const handleSelectBuild = (build: Build) => {
+    setSelectedBuild(build);
+  };
+
+  useEffect(() => {
```

Comment:
why do you need to set it to null on any change ?
you can change file selected on any select build instead